ui/core-react/src/core-react/messagebox/MessageBox.tsx

```diff
@@ -40,6 +40,7 @@ export interface MessageBoxProps extends CommonProps {
   /** List of [[DialogButtonDef]] objects specifying buttons and associated onClick events */
   buttonCluster: DialogButtonDef[];
   /** Title to show in title bar of dialog  */
+  // eslint-disable-next-line deprecation/deprecation
   title?: string | JSX.Element;
```

Comment:
Should we replace it with `React.ReactElement` or `React.JSX.Element` instead of ignoring it?

Reply:
> Did a quick pass didnt see any internal api usage? Most of it was just eslint disable deprecation comments? Shouldnt this pr fold in changes from #198 now?

Most of the internal APIs we were using were promoted in core. The changes are primarily in the test apps, getting rid of our reliance on MockRender. 
We should only fold in #198 if we can turn it off for the core-3x variant. Also, if it flags alpha and beta APIs it will cause too much noise.

Reply:
> Should we replace it with `React.ReactElement` or `React.JSX.Element` instead of ignoring it?

The error said to replace with React.JSX.Element, but that would not compile in the 3.x variant. I'll try React.ReactElement.